### DISCUSSION TITLE
Possible integration with crates.io

### DISCUSSION THREAD

**Discussion by alexpyattaev** [crev-dev/cargo-crev#733] [Ideas]
Has anyone considered/tried to build a possible mechanism for integration of crev into crates.io? It would make the crev system far more visible to the community, and may attract more reviewers. Or am I missing a point?

**Comment by dpc** [comment]
https://lib.rs/ already displays reviews it was able to find. crates.io could do the same, but it requires a champion to implement it, and AFAIK crates.io suffer from not enough contributors already. I'm also not sure how enthusiastic would they be about adding more complexity.

**Reply by alexpyattaev** [reply]
I had no idea lib.rs does that, though it seems they are not really doing a good job of it (it is pulling crev reviews in a very strange, non-obvious way). Given this, I suppose the best thing would be to help out the crates.io team, right?

Another thing is, given that crev is a web of trust, where should be the "entry point" into the web for something like crates.io? Like, it needs to trust some set of users implicitly, and pull from their trust chains, right?

**Reply by dpc** [reply]
> Another thing is, given that crev is a web of trust, where should be the "entry point" into the web for something like crates.io? 

Yeap, any system like that will need to figure out some way to gather the reviews. There was a project that was collecting such a "global WoT" to improve discovery. You might want to search the issue tracker.

> (it is pulling crev reviews in a very strange, non-obvious way)

I don't think there's any strange about it. AFAIK there isn't just many of such reviews, and maybe they could get more exposure, but it's debatable.

>  Given this, I suppose the best thing would be to help out the crates.io team, right?

Yeap. Most often, it's all about people that are willing to do the effort, yes. You might consider  contributing to https://gitlab.com/lib.rs first . I mean - it does have the integration done, so you'd just need to improve it if you see ways it could be done. For crates.io you'd probably first need to get some buy-in into the whole idea.

**Comment by alexpyattaev** [comment]
it seems that crev integration on crates.io/lib.rs raises more questions than it answers... in lib.rs case it is unclear where it attaches to the WoT... is there a good way to do this?
